docker/docker_image_dest.go

```diff
@@ -297,10 +299,15 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 
 		// Sanity checks:
 		if reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
-			logrus.Debugf("... Internal error: domain %s does not match destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
-			continue
+			if d.checkBlobEverywhere {
+				logrus.Debugf("... Ignoring domain %s does not match destination %s. checkBlobEverywhere enabled", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
```

Comment:
I prefer to abstract from the internals a bit to make this log easier to understand and suggest:

"... Cross repository blob check: ignoring domain mismatch between %s and %s"